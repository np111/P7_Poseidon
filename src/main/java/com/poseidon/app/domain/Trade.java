package com.poseidon.app.domain;

import javax.persistence.Entity;
import javax.persistence.Table;


@Entity
@Table(name = "trade")
public class Trade {
    // TODO: Map columns in data table TRADE with corresponding java fields
}
