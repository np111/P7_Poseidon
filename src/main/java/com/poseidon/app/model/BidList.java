package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class BidList {
    private Long id;
    private String account;
    private String type;
    private Double bidQuantity;
    private Double askQuantity;
    private Double bid;
    private Double ask;
    private String benchmark;
    private ZonedDateTime bidListDate;
    private String commentary;
    private String security;
    private String status;
    private String trader;
    private String book;
    private String creationName;
    private ZonedDateTime creationDate;
    private String revisionName;
    private ZonedDateTime revisionDate;
    private String dealName;
    private String dealType;
    private String sourceListId;
    private String side;
}
