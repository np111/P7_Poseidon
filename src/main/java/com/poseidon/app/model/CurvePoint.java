package com.poseidon.app.model;

import java.time.ZonedDateTime;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class CurvePoint {
    private Long id;
    private Long curveId;
    private ZonedDateTime asOfDate;
    private Double term;
    private Double value;
    private ZonedDateTime creationDate;
}
