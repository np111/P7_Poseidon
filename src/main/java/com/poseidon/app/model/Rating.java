package com.poseidon.app.model;

import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class Rating {
    private Long id;

    @NotBlank(message = "MoodysRating is mandatory")
    private String moodysRating;

    @NotBlank(message = "SandPRating is mandatory")
    private String sandPRating;

    @NotBlank(message = "FitchRating is mandatory")
    private String fitchRating;

    @NotNull(message = "Order is mandatory")
    private Byte orderNumber;
}
