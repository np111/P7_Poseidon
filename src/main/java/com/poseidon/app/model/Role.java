package com.poseidon.app.model;

public enum Role {
    USER,
    ADMIN,
}
