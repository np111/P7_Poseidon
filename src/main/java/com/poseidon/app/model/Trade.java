package com.poseidon.app.model;

import java.time.ZonedDateTime;
import javax.validation.constraints.DecimalMin;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Builder(builderClassName = "Builder")
@NoArgsConstructor
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Data
public class Trade {
    private Long id;

    @NotBlank(message = "Account is mandatory")
    private String account;

    @NotBlank(message = "Type is mandatory")
    private String type;

    @NotNull(message = "Buy Quantity is mandatory")
    @DecimalMin(value = "0", inclusive = false, message = "Buy Quantity must be positive")
    private Double buyQuantity;

    private Double sellQuantity;
    private Double buyPrice;
    private Double sellPrice;
    private ZonedDateTime tradeDate;
    private String security;
    private String status;
    private String trader;
    private String benchmark;
    private String book;
    private String creationName;
    private ZonedDateTime creationDate;
    private String revisionName;
    private ZonedDateTime revisionDate;
    private String dealName;
    private String dealType;
    private String sourceListId;
    private String side;
}
