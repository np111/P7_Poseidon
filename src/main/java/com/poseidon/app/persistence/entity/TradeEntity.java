package com.poseidon.app.persistence.entity;

import java.time.ZonedDateTime;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Table;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "trade")
@NoArgsConstructor
@Data
public class TradeEntity {
    public TradeEntity(String account, String type) {
        this.account = account;
        this.type = type;
        this.creationDate = ZonedDateTime.now();
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @Column(name = "account")
    private String account;

    @Column(name = "type")
    private String type;

    @Column(name = "buyQuantity")
    private Double buyQuantity;

    @Column(name = "sellQuantity")
    private Double sellQuantity;

    @Column(name = "buyPrice")
    private Double buyPrice;

    @Column(name = "sellPrice")
    private Double sellPrice;

    @Column(name = "tradeDate")
    private ZonedDateTime tradeDate;

    @Column(name = "security")
    private String security;

    @Column(name = "status")
    private String status;

    @Column(name = "trader")
    private String trader;

    @Column(name = "benchmark")
    private String benchmark;

    @Column(name = "book")
    private String book;

    @Column(name = "creationName")
    private String creationName;

    @Column(name = "creationDate")
    private ZonedDateTime creationDate;

    @Column(name = "revisionName")
    private String revisionName;

    @Column(name = "revisionDate")
    private ZonedDateTime revisionDate;

    @Column(name = "dealName")
    private String dealName;

    @Column(name = "dealType")
    private String dealType;

    @Column(name = "sourceListId")
    private String sourceListId;

    @Column(name = "side")
    private String side;
}
