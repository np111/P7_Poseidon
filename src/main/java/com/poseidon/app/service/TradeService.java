package com.poseidon.app.service;

import com.poseidon.app.model.Trade;
import com.poseidon.app.persistence.entity.TradeEntity;
import com.poseidon.app.persistence.mapper.TradeMapper;
import com.poseidon.app.persistence.repository.TradeRepository;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Service;

@RequiredArgsConstructor(onConstructor = @__(@Autowired))
@Service
@Scope("singleton")
public class TradeService extends CrudService<Trade, TradeEntity, Long> {
    private final @Getter TradeRepository repository;
    private final @Getter TradeMapper mapper;

    @Override
    protected void setModelId(Trade user, Long id) {
        user.setId(id);
    }
}
