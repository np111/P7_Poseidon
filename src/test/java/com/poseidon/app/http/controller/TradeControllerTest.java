package com.poseidon.app.http.controller;

import com.poseidon.app.model.Trade;
import com.poseidon.app.persistence.entity.TradeEntity;
import com.poseidon.app.service.TradeService;
import lombok.Getter;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.mock.mockito.MockBean;

@WebMvcTest(TradeController.class)
class TradeControllerTest extends AbstractCrudControllerTest<TradeService, Trade, TradeEntity, Long> {
    @MockBean
    private @Getter TradeService service;

    public TradeControllerTest() {
        super("/trade", Trade.class, Long.class);
    }

    @Override
    protected Trade getModelA() {
        return Trade.builder()
                .id(1L)
                .account("account")
                .type("type")
                .buyQuantity(2D)
                .build();
    }

    @Override
    protected Trade getModelB() {
        return Trade.builder()
                .id(535856L)
                .account("tcuLDhiU7w")
                .type("WeENe5bmtP")
                .buyQuantity(1.84388D)
                .build();
    }
}
