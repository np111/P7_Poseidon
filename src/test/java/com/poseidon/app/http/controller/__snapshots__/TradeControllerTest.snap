com.poseidon.app.http.controller.TradeControllerTest.add=[
-
> GET /trade/add
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Trade</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/trade/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"account\" class=\"col-sm-2 control-label\">Account</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"account\" placeholder=\"Account\" class=\"col-4\" name=\"account\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"type\" class=\"col-sm-2 control-label\">Type</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"type\" placeholder=\"Type\" class=\"col-4\" name=\"type\" value=\"\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"buyQuantity\" class=\"col-sm-2 control-label\">Buy Quantity</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"buyQuantity\" placeholder=\"Buy Quantity\" class=\"col-4\" name=\"buyQuantity\" value=\"\">"
"          "
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/trade/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add trade\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.TradeControllerTest.delete=[
-
> GET /trade/delete/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /trade/delete/535856
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /trade/list
""
]


com.poseidon.app.http.controller.TradeControllerTest.list=[
-
> GET /trade/list
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <div class=\"col-6\">"
"      <a href=\"/bidList/list\">Bid List</a>&nbsp;|&nbsp;"
"      <a href=\"/curvePoint/list\">Curve Points</a>&nbsp;|&nbsp;"
"      <a href=\"/rating/list\">Ratings</a>&nbsp;|&nbsp;"
"      <a href=\"/trade/list\">Trade</a>&nbsp;|&nbsp;"
"      <a href=\"/ruleName/list\">Rule</a>"
"    </div>"
"    <div class=\"col-6 text-right\">"
"      Logged in user: <b class=\"user\"> test-username </b>"
"      <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"        <input type=\"submit\" value=\"Logout\"/>"
"      </form>"
"    </div>"
"  </div>"
"  <div class=\"row\"><h2>Trade List</h2></div>"
"  <div class=\"row\">"
"    <a href=\"/trade/add\" class=\"btn btn-primary btn-sm\">Add New</a>"
"    <table class=\"table table-bordered\">"
"      <thead>"
"      <tr>"
"        <th>Id</th>"
"        <th>Account</th>"
"        <th>Type</th>"
"        <th>Buy Quantity</th>"
"        <th>Action</th>"
"      </tr>"
"      </thead>"
"      <tbody>"
"      <tr>"
"        <td>1</td>"
"        <td>account</td>"
"        <td>type</td>"
"        <td>2.0</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/trade/update/1\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/trade/delete/1\">Delete</a>"
"        </td>"
"      </tr>"
"      <tr>"
"        <td>535856</td>"
"        <td>tcuLDhiU7w</td>"
"        <td>WeENe5bmtP</td>"
"        <td>1.84388</td>"
"        <td style=\"width: 15%\" class=\"text-center\">"
"          <a href=\"/trade/update/535856\">Edit</a>&nbsp;|&nbsp;"
"          <a href=\"/trade/delete/535856\">Delete</a>"
"        </td>"
"      </tr>"
"      </tbody>"
"    </table>"
"  </div>"
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.TradeControllerTest.showUpdate=[
-
> GET /trade/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <title>Spring Boot</title>"
"</head>"
"<body>"
"<h3>Not Found Exception</h3>"
"<div>"
"  Logged in user: <b class=\"user\"> test-username </b>"
"  <form action=\"/app-logout\" method=\"POST\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"    <input type=\"submit\" value=\"Logout\"/>"
"  </form>"
"</div>"
"<p class=\"error\">Entity not found.</p>"
"</body>"
"</html>"
-
> GET /trade/update/535856
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <h2>Update Trade</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/trade/update/535856\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"account\" class=\"col-sm-2 control-label\">Account</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"account\" placeholder=\"Account\" class=\"col-4\" name=\"account\" value=\"tcuLDhiU7w\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"type\" class=\"col-sm-2 control-label\">Type</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"type\" placeholder=\"Type\" class=\"col-4\" name=\"type\" value=\"WeENe5bmtP\">"
"          "
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"buyQuantity\" class=\"col-sm-2 control-label\">Buy Quantity</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"buyQuantity\" placeholder=\"Buy Quantity\" class=\"col-4\" name=\"buyQuantity\" value=\"1.84388\">"
"          "
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/trade/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update trade\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
]


com.poseidon.app.http.controller.TradeControllerTest.update=[
-
> POST /trade/update/1
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
"  <div class=\"row\">"
"    <h2>Update Trade</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/trade/update/1\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"account\" class=\"col-sm-2 control-label\">Account</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"account\" placeholder=\"Account\" class=\"col-4\" name=\"account\" value=\"\">"
"          <p class=\"text-danger\">Account is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"type\" class=\"col-sm-2 control-label\">Type</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"type\" placeholder=\"Type\" class=\"col-4\" name=\"type\" value=\"\">"
"          <p class=\"text-danger\">Type is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"buyQuantity\" class=\"col-sm-2 control-label\">Buy Quantity</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"buyQuantity\" placeholder=\"Buy Quantity\" class=\"col-4\" name=\"buyQuantity\" value=\"\">"
"          <p class=\"text-danger\">Buy Quantity is mandatory</p>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/trade/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Update trade\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /trade/update/1
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /trade/list
""
]


com.poseidon.app.http.controller.TradeControllerTest.validate=[
-
> POST /trade/validate
< HTTP 200
< Content-Language: en
< Content-Type: text/html;charset=UTF-8
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
"<!DOCTYPE html>"
"<html xmlns=\"http://www.w3.org/1999/xhtml\">"
"<head>"
"  <meta charset=\"utf-8\"/>"
"  <title>Home</title>"
"  <link rel=\"stylesheet\" href=\"../../css/bootstrap.min.css\">"
"</head>"
"<body>"
"<div class=\"container\">"
""
"  <div class=\"row\">"
"    <h2>Add New Trade</h2>"
"  </div>"
""
"  <div class=\"row\">"
"    <form action=\"/trade/validate\" method=\"post\" class=\"form-horizontal\" style=\"width: 100%\"><input type=\"hidden\" name=\"_csrf\" value=\"12345678-903d-4d4e-8890-3e46ad33ce00\"/>"
"      <div class=\"form-group\">"
"        <label for=\"account\" class=\"col-sm-2 control-label\">Account</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"account\" placeholder=\"Account\" class=\"col-4\" name=\"account\" value=\"\">"
"          <p class=\"text-danger\">Account is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"type\" class=\"col-sm-2 control-label\">Type</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"type\" placeholder=\"Type\" class=\"col-4\" name=\"type\" value=\"\">"
"          <p class=\"text-danger\">Type is mandatory</p>"
"        </div>"
"      </div>"
"      <div class=\"form-group\">"
"        <label for=\"buyQuantity\" class=\"col-sm-2 control-label\">Buy Quantity</label>"
"        <div class=\"col-sm-10\">"
"          <input type=\"text\" id=\"buyQuantity\" placeholder=\"Buy Quantity\" class=\"col-4\" name=\"buyQuantity\" value=\"\">"
"          <p class=\"text-danger\">Buy Quantity is mandatory</p>"
"        </div>"
"      </div>"
""
""
"      <div class=\"form-group\">"
"        <div class=\"col-sm-12\">"
"          <a class=\"btn btn-danger btn-sm\" href=\"/trade/list\">Cancel</a>"
"          <input class=\"btn btn-primary btn-sm\" type=\"submit\" value=\"Add trade\">"
"        </div>"
"      </div>"
""
"    </form>"
"  </div>"
""
"</div>"
"</body>"
"</html>"
-
> POST /trade/validate
< HTTP 302
< Content-Language: en
< X-Content-Type-Options: nosniff
< X-XSS-Protection: 1; mode=block
< Cache-Control: no-cache, no-store, max-age=0, must-revalidate
< Pragma: no-cache
< Expires: 0
< X-Frame-Options: DENY
< Location: /trade/list
""
]